common/src/main/java/com/microsoft/identity/common/internal/util/AndroidKeyStoreUtil.java

```diff
@@ -447,11 +454,38 @@ public static synchronized SecretKey unwrap(@NonNull final byte[] wrappedKeyBlob
                 exception
         );
         if (exception instanceof InvalidKeyException) {
+            String ksMessage;
+            final String errorType;
+            final String ksNumericErrorCode;
+
+            // Check API Level before attempting to extract KeyStoreException details
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                final android.security.KeyStoreException keyStoreException = findKeyStoreException(exception);
+                if (keyStoreException != null) {
+                    ksMessage = keyStoreException.getMessage();
+                    if (ksMessage == null) {
+                        ksMessage = "Keystore exception found, no error message";
+                    }
+                    errorType = "KeyStoreException";
+                    ksNumericErrorCode = String.valueOf(keyStoreException.getNumericErrorCode());
+                } else {
+                    ksMessage = "No keystore exception found";
+                    errorType = "InvalidKeyException";
+                    ksNumericErrorCode = "";
+                }
+            } else {
+                ksMessage = "API Level below 33, keystore exception not available";
+                errorType = "InvalidKeyException";
+                ksNumericErrorCode = "";
+            }
+
```

Comment:
We can separate this in to a method

addAttributesForKeystoreExxception(invalidKeyException, attributesbuilder)

Reply:
Yeah i dont really like how this ended up looking either, i'll make a helper method